docs/middleware/op-submission.md

```diff
@@ -157,3 +157,21 @@ backend.use('apply', (request, next) => {
 
 {: .warn :}
 The `request.$fixup()` method may throw an error, which should be handled appropriately, usually by passing directly to the `next()` callback.
+
+## Comparing old snapshot version with new version
+
+Frequently, it becomes necessary to verify the changes made. This can be accomplished by leveraging two hooks, `apply` and `commit`, and creating a snapshot clone within the `apply` hook.
+
+```js
+backend.use('apply', (request, next) => {
+  request.snapshotBeforeApply = utils.clone(request.snapshot);
```

Comment:
I don't think we export `utils`, so should probably either:

 - change this to a generic `clone()` with a note about picking your favourite deep clone; or
 - just replace with `JSON.parse(JSON.stringify())`, which is [what this does anyway](https://github.com/share/sharedb/blob/dcef5491ca400826389c823a9ae2c53e2b02beac/lib/util.js#L98)